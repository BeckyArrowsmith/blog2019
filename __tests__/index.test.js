/* eslint-env jest */

import { shallow } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'

import App from '../pages/index.js'

describe('With Enzyme', () => {
    it('App renders without crashing', () => {
      shallow(<App />);
    });
})

describe('With Snapshot Testing', () => {
    it('App is rendered is rendered', () => {
        const component = renderer.create(<App />)
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})
