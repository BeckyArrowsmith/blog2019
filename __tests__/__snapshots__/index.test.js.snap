// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing App is rendered is rendered 1`] = `
<div>
  <header>
    <a
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "marginRight": 15,
        }
      }
    >
      Home
    </a>
    <a
      href="/about"
      onClick={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "marginRight": 15,
        }
      }
    >
      About
    </a>
  </header>
  <div>
    <article
      id="another-post"
    >
      <h2>
        <a
          href="http://localhost:3000/post?slug=another-post"
        >
          Another post
        </a>
      </h2>
      <p>
        Created at: 
        2019-09-02T15:44:40.520Z
      </p>
      <p>
        Updated at: 
        2019-09-03T17:45:09.743Z
      </p>
      <p>
        Another posts content
      </p>
      <section
        id="tags"
      >
        <h3>
          Tags:
        </h3>
        <ul>
          <li>
            <a
              href="JavaScript"
            >
              JavaScript
            </a>
          </li>
          <li>
            <a
              href="Testing"
            >
              Testing
            </a>
          </li>
        </ul>
      </section>
    </article>
    <article
      id="writing-unittests-with-jest-in-nextjs"
    >
      <h2>
        <a
          href="http://localhost:3000/post?slug=writing-unittests-with-jest-in-nextjs"
        >
          Writing unittests with Jest in Next.js
        </a>
      </h2>
      <p>
        Created at: 
        2019-07-30T20:08:58.693Z
      </p>
      <p>
        Updated at: 
        2019-07-31T20:39:42.280Z
      </p>
      <p>
        Content
      </p>
      <section
        id="tags"
      >
        <h3>
          Tags:
        </h3>
        <ul>
          <li>
            <a
              href="Testing"
            >
              Testing
            </a>
          </li>
        </ul>
      </section>
    </article>
  </div>
  <footer>
    <a
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "marginRight": 15,
        }
      }
    >
      link
    </a>
    <a
      href="/about"
      onClick={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "marginRight": 15,
        }
      }
    >
      link
    </a>
  </footer>
</div>
`;
